src/Numeric/Sundials/CVode.hs

```diff
@@ -197,6 +199,19 @@ solveC CConsts{..} CVars{..} log_env =
   }
 
   while (1) {
+     // The solver will run until it terminates or receive a signal to stop by the
+     // way of a non null value in *ptrSTop
+     // The signal is an exception from outside (see the solve function in
+     // Numeric/Sundials.hs
+    // Ensure proper memory barrier so that `stopFlag` is not optimised away as constant
```

Comment:
```suggestion
    // Ensure proper memory barrier.
    // This cannot be simply replaced by if(*ptrStop) because the compiler is free to consider ptrStop as a constant for the complete loop execution.
    // So instead, we use __atomic_load to force the load
```